### ISSUE TITLE
alert is not output (IE 11, JAWS)

### ISSUE THREAD

**Issue by JAWS-test** [w3c/aria-practices#1138]
https://w3c.github.io/aria-practices/examples/alert/alert.html

IE 11 is a bit difficult with role=alert, the example does not work with JAWS version 15 to 2019. It would be better to choose an implementation that also works with IE 11. 

Note: If the alert message has already been displayed and the button is pressed again, the output will also be correct with IE 11.

The example is nice international, but the language change within alert doesn't work with Chrome and Firefox either, so the output is a bit confusing...

**Comment by JAWS-test** [comment]
A ticket has also been added to JAWS: https://github.com/FreedomScientific/VFO-standards-support/issues/38

**Comment by JAWS-test** [comment]
I'd also like the ARIA practice to explain exactly which updates will issue an alert or a live region. Unfortunately I didn't find any information. It would be valuable to differentiate between
* what does the specification say
* what works in reality in different browsers and screen readers.

Background to the question: At least in IE 11 and Firefox different changes to the DOM and CSS lead to the output of alerts, e.g.
* Insert into DOM
* Remove from DOM
* Change visibility (display:none, hidden, aria-hidden)
* Changing the position via CSS (e.g.: left:100px)
* Change the color (e.g.: color:red)
* scale (e.g. width:100px)
* Change transparency (e.g. opacity:0;)
However, the output differs depending on whether these changes are made to the parent element, the element with role=alert, a direct child element, or a nested descendant element.
The output also varies depending on which Javascript method is used to insert element nodes into the DOM. 
Some of the test results are relatively new (JAWS 18, JAWS 2019), others quite old (JAWS 15).

**Comment by mcking65** [comment]
@JAWS-test commented:

> I'd also like the ARIA practice to explain exactly which updates will result in output of an alert or an other live region by screen reader. Unfortunately I didn't find any information. It would be valuable to differentiate between
> 
> * what does the specification say

I agree, the practices should cover this. I am adding this to issue#78 to be addressed in PR #1027. Once that content is written, it should be referenced from roles, states, and properties subsection of the alert pattern. 

> * what works in reality in different browsers and screen readers.

This is not the domain of the APG. It will eventually be covered by the w3c/aria-at project. APG says what to do based on the spec, not how well it works. the aria-at project will address how well it works in practice.

> Background of the question: At least in IE 11, Chrome and Firefox different changes to the DOM and CSS lead to the output of alerts (with JAWS), e.g.
> 
> * Insert into DOM
> * Remove from DOM
> * add role=alert to an existing node
> * Change visibility (display:none, hidden, aria-hidden)
> * Change the position (e.g.: left:100px)
> * Change the color (e.g.: color:red)
> * Change the size (e.g. width:100px)
> * Change transparency (e.g. opacity:0;)
> 
> However, the output differs depending on whether these changes are made to the parent element, the element with role=alert, a direct child element, or a nested descendant element.
> The output also varies depending on which Javascript method is used to insert element nodes into the DOM.

I'll add this list to issue #78.

**Comment by mcking65** [comment]
@JAWS-test, since we do not change examples to accommodate assistive technology bugs, I am closing this issue as not for practices.

However, I have updated issue #78 and created issue #1139 to address the other excellent feedback you have provided -- thank you!

**Comment by JAWS-test** [comment]
@mcking65: Thank you very much.
